cmake/JuliaCCompiler.cmake

```diff
@@ -0,0 +1,84 @@
+#
+
+# libc++ is standard on OS X 10.9, but not for earlier releases
+jl_option(USE_LIBCPP "Use libc++" Off)
+# assume we don't have LIBSSP support in our compiler,
+# will enable later if likely true
+set(HAVE_SSP Off)
+
+jl_set_option(USEICC Off)
+jl_set_option(USEGCC Off)
+jl_set_option(USECLANG Off)
+jl_set_option(USEMSVC Off)
+
+push_c_flags(CMAKE_CXX_FLAGS -std=c++11)
+
+if(CMAKE_C_COMPILER_ID STREQUAL "AppleClang" OR
+    CMAKE_C_COMPILER_ID STREQUAL "Clang")
+  jl_set_option(USECLANG On)
+  add_definitions(-pipe -fPIC)
+  push_c_flags(CMAKE_C_FLAGS -fno-strict-aliasing -D_FILE_OFFSET_BITS=64)
+  push_c_flags(CMAKE_CXX_FLAGS -fno-rtti)
+  push_c_flags(CMAKE_C_FLAGS_DEBUG -O0 -g -DJL_DEBUG_BUILD
+    -fstack-protector-all)
+  push_c_flags(CMAKE_CXX_FLAGS_DEBUG -O0 -g -DJL_DEBUG_BUILD
+    -fstack-protector-all)
+  push_c_flags(CMAKE_C_FLAGS_RELEASE -O3 -g)
+  push_c_flags(CMAKE_CXX_FLAGS_RELEASE -O3 -g)
+  if(APPLE)
+    if(USE_LIBCPP)
+      add_definitions(-stdlib=libc++ -mmacosx-version-min=10.7)
+    else()
+      add_definitions(-mmacosx-version-min=10.6)
+    endif()
+    add_definitions(-D_LARGEFILE_SOURCE -D_DARWIN_USE_64_BIT_INODE=1)
+  endif()
+elseif(CMAKE_C_COMPILER_ID STREQUAL "GNU")
+  jl_set_option(USEGCC On)
+  add_definitions(-pipe -fPIC)
```

Comment:
I see. These compiler flags are far from complete. I'm planing to do a more careful port of the flags after the infrastructure is there.